lib/kubeclient/informer.rb

```diff
@@ -34,22 +34,20 @@ def start_worker
           ensure
             sleep(1) # do not overwhelm the api-server if we are somehow broken
           end
-          break if @stopped
         end
       end
       sleep(0.01) until @cache || @stopped
     end
 
     def stop_worker
-      @stopped = true
-      [@waiter, @worker].compact.each do |thread|
-        begin
-          thread.run # cancel sleep so either the loop sleep or the timeout sleep are interrupted
-        rescue ThreadError
-          # thread was already dead
-        end
-        thread.join
+      @stopped = true # mark that any threads should be stopped if it is running
+      begin
+        Thread.pass
+        @worker.run # interrupt potential loop sleep
```

Comment:
This should not be necessary, since at most it will avoid the `sleep 1` in the loop. Condition on `@stopped` is suficient.

Reply:
avoiding a `sleep 1` could be useful to make the tests finish quickly ... unless they stub all sleeps